julia/LibCEED.jl/src/LibCEED.jl

```diff
@@ -145,11 +149,104 @@ include("Request.jl")
+            The version of the libCEED library is $libceed_version."
+            """)
+    end
     @require CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba" include("Cuda.jl")
     set_globals()
 end
 
+"""
+    ceedversion()
+
+Returns a `VersionNumber` corresponding to the version of the libCEED library currently used.
+"""
+function ceedversion()
+    major = Ref{Cint}()
+    minor = Ref{Cint}()
+    patch = Ref{Cint}()
+    release = Ref{Bool}()
+    C.CeedGetVersion(major, minor, patch, release)
+    return VersionNumber(major[], minor[], patch[])
+end
+
+"""
+    isrelease()
+
+Returns true if the libCEED library is a release build, false otherwise.
+"""
+function isrelease()
+    major = Ref{Cint}()
+    minor = Ref{Cint}()
+    patch = Ref{Cint}()
+    release = Ref{Bool}()
+    C.CeedGetVersion(major, minor, patch, release)
+    return release[]
+end
+
+"""
+    set_libceed_path!(path::AbstractString)
+
+Sets the path of the libCEED dynamic library. `path` should be the absolute path to the library
+file. This function sets the library path as a _preference_ associated with the currently active
+environment. Changes will take effect after restarting the Julia session. See the Preferences.jl
+documentation for more information.
+"""
+function set_libceed_path!(path::AbstractString)
+    set_preferences!(C.libCEED_jll, "libceed_path" => path; force=true)
+    @info("""
+          Setting the libCEED library path to $path.
+          Restart the Julia session for changes to take effect.
+          """)
+end
+
+"""
+    unset_libceed_path!()
+
+Unsets the path of the libCEED dynamic library (i.e. undoes a call to [`set_libceed_path!`](@ref)).
+This function deletes the library path preference associated with the currently active environment.
+Changes will take effect after restarting the Julia session. See the Preferences.jl documentation
+for more information.
+"""
+function unset_libceed_path!()
+    delete_preferences!(Preferences.get_uuid(C.libCEED_jll), "libceed_path"; force=true)
+    @info("""
+          Deleting preference for libCEED library path.
+          Restart the Julia session for changes to take effect.
+          """)
+end
+
+"""
+    use_prebuilt_libceed!()
+
+Indicates that the prebuilt version of the libCEED library (bundled with libCEED_jll) should be
+used. This function *clears* the library path preference associated with the currently active
+environment. Changes will take effect after restarting the Julia session. See the Preferences.jl
+documentation for more information. If a depot-wide Overrides.toml file is present (e.g. in
+~/.julia/artifacts, a different libCEED library may still be used).
+"""
+function use_prebuilt_libceed!()
+    set_preferences!(C.libCEED_jll, "libceed_path" => nothing; force=true)
+    @info("""
+          Using the prebuilt libCEED binary.
+          Restart the Julia session for changes to take effect.
+          """)
+end
```

Comment:
Do we need both modes? It sounds like they're similar and it may be confusing which should be used.

Reply:
Probably not… they are similar, and in most cases will give the same result.

The difference is that `use_prebuilt_libceed!` will "force" the library to use the prebuilt library, whereas `unset_libceed_path!` will tell LibCEED.jl that the user has no preference. Usually, this will also mean using the prebuilt library, but there are situations where this default could be changed somewhere else (for example, if there is a shared installation of libCEED and LibCEED.jl, then the default library path could be changed for everyone using this shared installation).

Reply:
Would it be idiomatic to consolidate something like this?
```julia
set_libceed_path!("path/to/libceed.so")
set_libceed_path!(:prebuilt) # Always use pre-built, regardless of environment
set_libceed_path!(:default)  # Try environment, then fall back to prebuilt
```

Reply:
To me it seems a little strange to call `set_libceed_path!` with an argument that isn't a path.

`use_prebuilt_libceed!` is more expressive in my opinion. We can probably just get rid of `unset_libceed_path!`, the user can always unset the preferences manually (either by calling `set_preferences!` or just editing the preferences TOML file).

Do you think that would make sense?

Reply:
Yeah, that's fine. My thinking was that when you give a user one lever, they `set` it wherever they want and the implication is that the history doesn't matter. But when you give them several levers, it can be hard to tell whether they interact, whether there can be any order/history dependence between `use_prebuilt` and `set`. Similar for a key in the TOML. If you set one key, then it always has a unique and well-specified state after parsing. If there are several keys, then you need some logic to determine whether the state is invalid and whether to report the error or resolve it according to some rule.

Reply:
Yet another option is to implement the interface you proposed, and keep `use_prebuilt_libceed!` as an alias. The docstring would be something like:
```
    use_prebuilt_libceed!()

Indicates that the prebuilt version of the libCEED library (bundled with libCEED_jll) should be
used. Equivalent to `set_libceed_path!(:prebuilt)`.
```

Reply:
It's more than one way to do it, but entirely clear about effects/side-effects and tab-completes better than `set_libceed_path!(:prebuilt)`.

Reply:
Agreed, done.